docs/docsite/rst/network/user_guide/platform_junos.rst

```diff
@@ -0,0 +1,113 @@
+| |                          | |                                                      | | text: ``result[1].interface-information[0].physical-interface[0].name[0].data foo lo0``          |
+| |                          | |                                                      | | xml: ``result[1].rpc-reply.interface-information[0].physical-interface[0].name[0].data foo lo0`` |
++----------------------------+--------------------------------------------------------+----------------------------------------------------------------------------------------------------+
+
+For legacy playbooks, Ansible still supports ``ansible_connection: local`` on all JUNOS modules. We recommend modernizing to use ``ansible_connection: netconf`` or ``ansible_connection: network_cli`` as soon as possible.
+
+Using CLI in Ansible 2.5
+================================================================================
+
+Example CLI ``group_vars/junos.yml``
+------------------------------------
+
+.. code-block:: yaml
+
+   ansible_connection: network_cli
+   ansible_network_os: junos
+   ansible_user: myuser
+   ansible_ssh_pass: !vault...
+   ansible_ssh_common_args: '-o ProxyCommand="ssh -W %h:%p -q bastion01"'
+
+
+- If you are using SSH keys (including an ssh-agent) you can remove the ``ansible_ssh_pass`` configuration.
+- If you are accessing your host directly (not through a bastion/jump host) you can remove the ``ansible_ssh_common_args`` configuration.
+- If you are accessing your host through a bastion/jump host, you cannot include your SSH password in the ``ProxyCommand`` directive. To prevent secrets from leaking out (for example in ``ps`` output), SSH does not support providing passwords via environment variables.
+
+Example CLI Task
+----------------
+
+.. code-block:: yaml
+
+   - name: Backup current switch config (junos)
+     junos_config:
+       backup: yes
+     register: backup_junos_location
+     when: ansible_network_os == 'junos'
+
+
+Using NETCONF in Ansible 2.5
+================================================================================
+
+Enabling NETCONF
+----------------
+
+Before you can use NETCONF to connect to a switch, you must:
+
+- install the ``ncclient`` python package on your control node(s) with ``pip install ncclient``
+- enable NETCONF on the Junos OS device(s)
+
+To enable NETCONF on a new switch via Ansible, use the ``junos_netconf`` module via the CLI connection. Set up group_vars/junos.yml just like in the CLI example above, then run a playbook task like this:
+
+.. code-block:: yaml
+
+   - name: Enable NETCONF
+      junos_netconf:
+      when: ansible_network_os == 'junos'
+
+Once NETCONF is enabled, change your ``group_vars/junos.yml`` to use the NETCONF connection.
+
+Example NETCONF ``group_vars/junos.yml``
+----------------------------------------
+
+.. code-block:: yaml
+
+   ansible_connection: netconf
+   ansible_network_os: junos
+   ansible_user: myuser
+   ansible_pass: !vault | 
```

Comment:
`ansible_ssh_pass`

Reply:
Thanks for catching that. Fixed in 793140ded6f7f5357d413d17ec82db9b7d4bed7f.